src/routers/Gateway.sol

```diff
@@ -275,6 +309,56 @@ contract ConnectorGateway {
             (uint64 poolId, bytes16 trancheId, address destinationAddress, uint128 amount) =
                 ConnectorMessages.parseTransferTrancheTokens20(_msg);
             connector.handleTransferTrancheTokens(poolId, trancheId, destinationAddress, amount);
+        } else if (ConnectorMessages.isExecutedDecreaseInvestOrder(_msg)) {
+            (
+                uint64 poolId,
+                bytes16 trancheId,
+                bytes32 investor,
+                uint128 currency,
+                uint128 currencyPayout,
+                uint128 remainingInvestOrder
+            ) = ConnectorMessages.parseExecutedDecreaseInvestOrder(_msg);
+            connector.handleExecutedDecreaseInvestOrder(
+                poolId, trancheId, investor, currency, currencyPayout, remainingInvestOrder
+            );
+        } else if (ConnectorMessages.isExecutedDecreaseRedeemOrder(_msg)) {
+            (
+                uint64 poolId,
+                bytes16 trancheId,
+                bytes32 investor,
+                uint128 currency,
+                uint128 trancheTokensPayout,
+                uint128 remainingRedeemOrder
+            ) = ConnectorMessages.parseExecutedDecreaseRedeemOrder(_msg);
+            connector.handleExecutedDecreaseRedeemOrder(
+                poolId, trancheId, investor, currency, trancheTokensPayout, remainingRedeemOrder
+            );
+        } else if (ConnectorMessages.isExecutedCollectInvest(_msg)) {
+            (
+                uint64 poolId,
+                bytes16 trancheId,
+                bytes32 investor,
+                uint128 currency,
+                uint128 currencyPayout,
```

Comment:
currencyInvested instead of currencyPayout

Reply:
I think we wanted the message parameters to be consistent between cent chain and solidity, so would not rename this here

Reply:
There were long discussions about these naming decisions in https://github.com/centrifuge/centrifuge-chain/pull/1363